src/voice/VoiceRecording.ts

```diff
@@ -80,18 +100,34 @@ export class VoiceRecording extends EventEmitter implements IDestroyable {
         // it makes the time domain less than helpful.
         this.recorderFFT.fftSize = 64;
 
-        // We use an audio processor to get accurate timing information.
-        // The size of the audio buffer largely decides how quickly we push timing/waveform data
-        // out of this class. Smaller buffers mean we update more frequently as we can't hold as
-        // many bytes. Larger buffers mean slower updates. For scale, 1024 gives us about 30Hz of
-        // updates and 2048 gives us about 20Hz. We use 1024 to get as close to perceived realtime
-        // as possible. Must be a power of 2.
-        this.recorderProcessor = this.recorderContext.createScriptProcessor(1024, CHANNELS, CHANNELS);
+        // Set up our worklet. We use this for timing information and waveform analysis: the
+        // web audio API prefers this be done async to avoid holding the main thread with math.
+        const mxRecorderWorkletPath = document.body.dataset.vectorRecorderWorkletScript;
+        if (!mxRecorderWorkletPath) {
+            throw new Error("Unable to create recorder: no worklet script registered");
+        }
+        await this.recorderContext.audioWorklet.addModule(mxRecorderWorkletPath);
+        this.recorderWorklet = new AudioWorkletNode(this.recorderContext, WORKLET_NAME);
```

Comment:
It seems like the `AudioWorkletNode` is not supported by Safari whereas the `ScriptProcessorNode` was.
Is there a fallback or progressive enhancement setup somewhere that I might not be aware of?

Reply:
ugh, I'll make note of this and get it properly tested in Safari in a future PR. I'm not confident it works properly before this change anyways.

Reply:
We'll definitely need some plan to support Safari before this can be enabled by default.